features/include/pcl/features/impl/feature.hpp

```diff
@@ -143,10 +143,11 @@ pcl::Feature<PointInT, PointOutT>::initCompute ()
     {
       search_parameter_ = search_radius_;
       // Declare the search locator definition
-      int (KdTree::*radiusSearchSurface)(const PointCloudIn &cloud, int index, double radius,
-                                         std::vector<int> &k_indices, std::vector<float> &k_distances,
-                                         unsigned int max_nn) const = &pcl::search::Search<PointInT>::radiusSearch;
-      search_method_surface_ = boost::bind (radiusSearchSurface, boost::ref (tree_), _1, _2, _3, _4, _5, 0);
+      search_method_surface_ = [this] (const PointCloudIn &cloud, int index, double radius,
+                                       std::vector<int> &k_indices, std::vector<float> &k_distances)
```

Comment:
The explicit types are mandatory here. The were passing compilation but I was getting a number of tests segfaulting.

Reply:
Did you use `auto` or `auto&` (where appropriate)?

Reply:
No. I just noticed it now. But then I raise the question, if I will need to check which type is required by the function object, isn't it better to adopt it's signature explicitly in the first place.

Reply:
> I feel that for these function objects the use of auto has more shortcomings 

I agree, when we need to create a functor object with a particular signature it makes sense to be explicit.

Reply:
I will complete the rest of this transition applying the explicit types then.